src/main/java/com/rocketden/main/service/ProblemService.java

```diff
@@ -83,6 +86,71 @@ public ProblemDto getProblem(String problemId) {
         return ProblemMapper.toDto(problem);
     }
 
+    public ProblemDto editProblem(String problemId, ProblemDto updatedProblem) {
+        Problem problem = repository.findProblemByProblemId(problemId);
+
+        if (problem == null) {
+            throw new ApiException(ProblemError.NOT_FOUND);
+        }
+
+        if (updatedProblem.getName() == null || updatedProblem.getDescription() == null
+                || updatedProblem.getDifficulty() == null
+                || updatedProblem.getProblemInputs() == null
+                || updatedProblem.getOutputType() == null) {
+            throw new ApiException(ProblemError.EMPTY_FIELD);
+        }
+
+        if (updatedProblem.getDifficulty() == ProblemDifficulty.RANDOM) {
+            throw new ApiException(ProblemError.BAD_DIFFICULTY);
+        }
+
+        // Ensure that the user entered valid inputs and outputs for the problem
+        for (ProblemTestCaseDto input : updatedProblem.getTestCases()) {
+            validateGsonParseable(input.getInput(), updatedProblem.getProblemInputs());
+            validateGsonParseable(input.getOutput(), updatedProblem.getOutputType());
+        }
+
+        problem.setName(updatedProblem.getName());
+        problem.setDescription(updatedProblem.getDescription());
+        problem.setDifficulty(updatedProblem.getDifficulty());
+        problem.setOutputType(updatedProblem.getOutputType());
+
+        problem.getProblemInputs().clear();
+        for (ProblemInputDto problemInput : updatedProblem.getProblemInputs()) {
+            if (problemInput != null) {
+                problem.addProblemInput(ProblemMapper.toProblemInputEntity(problemInput));
+            } else {
+                throw new ApiException(ProblemError.BAD_INPUT);
+            }
```

Comment:
```suggestion
            problem.addProblemInput(ProblemMapper.toProblemInputEntity(problemInput));
```
(I believe you already have this check in `validateGsonParseable`, so I don't think there's a need for it here.)